### PR TITLE
search by author

### PR DESCRIPTION
**Pull request by neuromusic**
This is an attempt at implementing #433

### CODE REVIEW

File: frontend/src/context/plugin.tsx [hunk 1]
```diff
@@ -176,7 +176,7 @@ export function usePluginMetadata() {
       value:
         plugin?.authors && isArray(plugin.authors)
           ? plugin.authors
-              ?.map((author) => author.name)
+              ?.flatMap((author) => author.name.split(/[,;&]+/))
```
Comment (neuromusic) [inline frontend/src/context/plugin.tsx:179]: @codemonkey800 even easier than I expected

File: frontend/src/components/MetadataList/MetadataListSearchItem.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+/**
+ * Component for rendering text items in metadata lists.
+ */
+export function MetadataListSearchItem({ children, className }: Props) {
+  const plausible = usePlausible();
+
+  // not sure how to get text from children?
+  const url = 'https://napari-hub.org/?search=' + encodeURIComponent(children);
```
Comment (neuromusic) [inline frontend/src/components/MetadataList/MetadataListSearchItem.tsx:22]: @codemonkey800 I probably shouldn't hardcode the domain here... what's the best way to get this to populate with the hostname? (e.g. localhost:8080 etc)

File: frontend/src/context/plugin.tsx (hunk 1 above)
Comment (codemonkey800) [reply frontend/src/context/plugin.tsx:179]: amazing, i didn't know you could pass a regex for split lol

File: frontend/src/components/MetadataList/MetadataListSearchItem.tsx (hunk 2 above)
Comment (codemonkey800) [reply frontend/src/components/MetadataList/MetadataListSearchItem.tsx]: you could probably use a forward slash `/?search=`

Comment (neuromusic) [reply frontend/src/components/MetadataList/MetadataListSearchItem.tsx]: of course 🤦

**Comment by neuromusic** [comment]
there's actually a new plugin that throws a wrench in the implementation here: https://www.napari-hub.org/plugins/recOrder-napari, for which the (valid) author is "Computational Microscopy Platform, CZ Biohub"

the current implementation will parse this and assign two authors to the plugin: "Computational Microscopy Platform" and "CZ Biohub", which is NOT what we want

further, since I'm currently parsing this as the data model level, this could have downstream effects (say, if/when we add filter-by-author)

**Pull request closed without merging**